File: src/main/java/fi/aalto/cs/apluscourses/intellij/notifications/UrlRenderingErrorNotification.java
package fi.aalto.cs.apluscourses.intellij.notifications;

import com.intellij.notification.Notification;
import com.intellij.notification.NotificationType;
import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
import org.jetbrains.annotations.NotNull;

import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getAndReplaceText;
import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;

public class UrlRenderingErrorNotification extends Notification {

  @NotNull
  private final Exception exception;

  /**
   * Construct a notification informing the user that an error occurred while attempting to render
   * a submission.
   */
  public UrlRenderingErrorNotification(@NotNull Exception exception) {
    super(
        PluginSettings.A_PLUS,
        getText("notification.UrlRenderingErrorNotification.title"),
        getAndReplaceText("notification.UrlRenderingErrorNotification.content",
            exception.getMessage()),
        NotificationType.ERROR
    );
    this.exception = exception;
  }

  @NotNull
  public Exception getException() {
    return exception;
  }

}

File: src/main/kotlin/fi/aalto/cs/apluscourses/notifications/UrlRenderingErrorNotification.kt
package fi.aalto.cs.apluscourses.notifications

import com.intellij.notification.Notification
import com.intellij.notification.NotificationType
import fi.aalto.cs.apluscourses.MyBundle
import fi.aalto.cs.apluscourses.services.PluginSettings

class UrlRenderingErrorNotification
/**
 * Construct a notification informing the user that an error occurred while attempting to render
 * a submission.
 */(val exception: Exception) : Notification(
    PluginSettings.A_PLUS,
    MyBundle.message("notification.UrlRenderingErrorNotification.title"),
    MyBundle.message(
        "notification.UrlRenderingErrorNotification.content",
        exception.message!!
    ),
    NotificationType.ERROR
)
